### DISCUSSION TITLE
How to style external component with Styled Components?

### DISCUSSION THREAD

**Discussion by nastiche** [neuefische/questions#626] [Web] [answered]
I imported an external component ReactTags from the react-tag-input library. It works very well, but I can't find a way to style it with Styled Components. The ReactTags component has attribute classNames (see first screenshot), but I don't know how to access those class names with Styled Components:

![Screenshot 2023-06-19 at 14 35 10](https://github.com/neuefische/questions/assets/97537490/3f19d467-2db3-486c-9f81-e4ac9c0912dd)

By now I don't have any style in my ReactTags component, so it's displayed with default style in my app:

![Screenshot 2023-06-19 at 14 36 24](https://github.com/neuefische/questions/assets/97537490/8f31f35b-88ca-4221-bada-ea73f5e856ac)

Is there a standard way to apply Styled Components method to external components?

I tried to style these ways, but it didn't work:

first way:

const TagEditor = styled(ReactTags)`
.ReactTags__remove {
    background-color: #ff0000;
    color: #ffffff;
    border: none;
    padding: 0.25rem;
    cursor: pointer;
  }
`

second way:

const TagEditor = styled(ReactTags)`
  &.tagInputField {
    background-color: red;
  }
  &.remove {
    background-color: red;
  }
`;

Thank you in advance,
Nasti

**Comment by mpagels** [comment]
There is a very good explanation and example in the [Readme](https://github.com/react-tags/react-tags#styling) of the project.
If you need any help with that, pls let me know.

**Reply by nastiche** [reply]
Hi Martin, I read the document, but didn't find anything concerning Styled Components. It seems like a very mysterious thing ))) I also tried lots of ways suggested on StackOverFlow, but nothing works unfortunately. Probably I will have to create a CSS file instead of using Styled Components approach.

**Comment by djfarly** [comment, marked as answer]
Id suggest you do the following:

```js
const MyTagsWrapper = styled.div`
  .react-tags__remove {
    background-color: #ff0000;
    color: #ffffff;
    border: none;
    padding: 0.25rem;
    cursor: pointer;
  }

 .react-tags__something-else {
   color: hotpink;
 }
`;
```

then when you render:

```js
return (
  <MyTagsWrapper>
    <ReactTags your={props} go={here} />
  </MyTagsWrapper>
);
```

This will apply the styles to any element with the class name that is a child of MyTagsWrapper. Of course you can choose any name you want :)

**Reply by nastiche** [reply]
Jan, you definitely made my day! It worked instantly...I spent like 9 hours on this one thing today ))) 

So I used the class names from the react-tag-input library:

![Screenshot 2023-06-19 at 18 13 47](https://github.com/neuefische/questions/assets/97537490/9ffbc683-191f-456d-9d1d-0b74725aec5a)
![Screenshot 2023-06-19 at 18 10 33](https://github.com/neuefische/questions/assets/97537490/d08d06b0-7bd8-4ff6-a5c6-90b67c4b361f)
![Screenshot 2023-06-19 at 18 15 08](https://github.com/neuefische/questions/assets/97537490/cf498f3e-3a64-4705-9b4c-1394e2071866)

Thank you so much!

**Reply by djfarly** [reply]
Happy that it works 🙌

**Reply by nastiche** [reply]
One last question: did I describe the issue in a right way? Like did you understand immediately what I struggled with? Just wondering, because it was my first issue post...
